samples/LowAllocationWebServer/LowAllocationWebServerLibrary/OwnedBuffer.cs

```diff
@@ -120,12 +127,12 @@ protected override void Dispose(bool disposing)
 
         public override Span<byte> GetSpan(int index, int length)
         {
-            if (IsDisposed) ThrowObjectDisposed();
             return Span.Slice(index, length);
         }
 
         protected override bool TryGetArrayInternal(out ArraySegment<byte> buffer)
         {
+            if (IsDisposed) ThrowObjectDisposed();
```

Comment:
Having `ThrowObjectDisposed` and `ThrowIndexOutOfRange` on a public abstract class is bad, use `ThrowHelper`